services/clusters-keeper/src/simcore_service_clusters_keeper/data/docker-compose.yml

```diff
@@ -16,28 +34,62 @@ services:
       LOG_LEVEL: ${LOG_LEVEL:-WARNING}
       SIDECAR_COMP_SERVICES_SHARED_FOLDER: ${SIDECAR_COMP_SERVICES_SHARED_FOLDER:-/home/scu/computational_shared_data}
       SIDECAR_COMP_SERVICES_SHARED_VOLUME_NAME: computational_shared_data
-    networks:
-      - default
     deploy:
       mode: global
+      placement:
+        constraints:
+          - "node.role==worker"
 
-  dask-scheduler:
-    image: ${DOCKER_REGISTRY:-itisfoundation}/dask-sidecar:${DOCKER_IMAGE_TAG}
+  autoscaling:
+    image: ${DOCKER_REGISTRY:-itisfoundation}/autoscaling:${DOCKER_IMAGE_TAG}
+    dns: 8.8.8.8 # needed to access internet from AWS
     init: true
     hostname: "{{.Node.Hostname}}-{{.Service.Name}}-{{.Task.Slot}}"
     environment:
-      DASK_START_AS_SCHEDULER: 1
+      DASK_MONITORING_URL: tcp://dask-scheduler:8786
+      EC2_ACCESS_KEY_ID: ${EC2_CLUSTERS_KEEPER_ACCESS_KEY_ID}
+      EC2_ENDPOINT: ${EC2_CLUSTERS_KEEPER_ENDPOINT}
+      EC2_INSTANCES_ALLOWED_TYPES: ${WORKERS_EC2_INSTANCES_ALLOWED_TYPES}
+      EC2_INSTANCES_AMI_ID: ${WORKERS_EC2_INSTANCES_AMI_ID}
+      EC2_INSTANCES_CUSTOM_BOOT_SCRIPTS: ${WORKERS_EC2_INSTANCES_CUSTOM_BOOT_SCRIPTS}
+      EC2_INSTANCES_KEY_NAME: ${WORKERS_EC2_INSTANCES_KEY_NAME}
+      EC2_INSTANCES_MAX_START_TIME: ${WORKERS_EC2_INSTANCES_MAX_START_TIME}
+      EC2_INSTANCES_MAX_INSTANCES: ${WORKERS_EC2_INSTANCES_MAX_INSTANCES}
+      EC2_INSTANCES_NAME_PREFIX: ${EC2_INSTANCES_NAME_PREFIX}
+      EC2_INSTANCES_SECURITY_GROUP_IDS: ${WORKERS_EC2_INSTANCES_SECURITY_GROUP_IDS}
+      EC2_INSTANCES_SUBNET_ID: ${WORKERS_EC2_INSTANCES_SUBNET_ID}
+      EC2_REGION_NAME: ${EC2_CLUSTERS_KEEPER_REGION_NAME}
+      EC2_SECRET_ACCESS_KEY: ${EC2_CLUSTERS_KEEPER_SECRET_ACCESS_KEY}
+      LOG_FORMAT_LOCAL_DEV_ENABLED: 1
       LOG_LEVEL: ${LOG_LEVEL:-WARNING}
-    ports:
-      - 8786:8786
-      - 8787:8787
-    networks:
-      - default
+      REDIS_HOST: redis
+      REDIS_PORT: 6379
+    volumes:
+      - "/var/run/docker.sock:/var/run/docker.sock"
+    deploy:
+      placement:
+        constraints:
+          - "node.role==manager"
+
+  redis:
+    # NOTE: currently autoscaling requires redis to run
```

Comment:
Q: if this service restarts does autoscaling continue to work?

Reply:
maybe. but since the background task restarts indefinitely, then yes.